File: LeetCode/53. Maximum Subarray/solution.go
func maxSubArray(nums []int) int {
	sum, maxS := nums[0], nums[0]
	for _, v := range nums[1:] {
		if sum < 0 {
			sum = v
		} else {
			sum += v
		}
		if maxS < sum {
			maxS = sum
		}
	}
	return maxS
}

File: LeetCode/53. Maximum Subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current = nums[0]
        result = nums[0]
        for i in range(1, len(nums)):
            current = max(nums[i], current + nums[i])
            result = max(result, current)
        return result